### PR TITLE
Fixes #16909: Escape password having special character '@'

### PR DESCRIPTION
**Pull request by anandkumaragrawal**
Escape password having special character '@'

### CODE REVIEW

File: app/lib/katello/util/http_proxy.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module Katello
   module Util
     module HttpProxy
       def proxy_uri
-        URI("#{proxy_scheme}://#{proxy_user_info}@#{proxy_host}:#{proxy_port}").to_s if proxy_host
+        CGI.escape("#{proxy_scheme}://#{proxy_user_info}@#{proxy_host}:#{proxy_port}").to_s if proxy_host
```
Comment (jlsherrill) [inline app/lib/katello/util/http_proxy.rb]: Does this actually fix the problem? 
```
> CGI.escape("http://foo:bar@example.com:3000")
=> "http%3A%2F%2Ffoo%3Abar%40example.com%3A3000"
```

I don't think that will work

**Comment by jlsherrill** [comment]
@anandkumaragrawal hey, this came up as a priority for a user and i was assigned to work on it, looking it to it, it was actually a lot more complicated than we originally discussed.  I've opened my pr here: https://github.com/Katello/katello/pull/7977

File: app/lib/katello/util/http_proxy.rb (hunk 1 above)
Comment (anandkumaragrawal) [reply app/lib/katello/util/http_proxy.rb]: I have reviewed this again, and this doesn't work. Feel free to remove this PR from the bug.

Thanks

**Comment by jlsherrill** [comment]
closing, thanks @anandkumaragrawal

**Pull request closed without merging**